src/pool/mod.rs

```diff
@@ -8,6 +8,41 @@ use crate::{
 
 use std::{fmt, marker::PhantomData};
 
+/// A lock-free concurrent object pool.
+///
+/// Slabs provide pre-allocated storage for many instances of a single type. But when working with
+/// heap allocated objects, the advantages of a Slab are lost as the memory allocated for the
+/// object is freed when the object is removed from the Slab. With a pool, we can instead re-use
```

Comment:
Nit: let's either spell slab lowercase when referring to slabs in the generic sense
```suggestion
/// Slabs provide pre-allocated storage for many instances of a single type. But when working with
/// heap allocated objects, the advantages of a slab are lost as the memory allocated for the
/// object is freed when the object is removed from the slab. With a pool, we can instead re-use
```
or, if we're referring specifically to the `Slab` type in this crate, we should say

```suggestion
/// [`Slab`]s provide pre-allocated storage for many instances of a single type. But when working with
/// heap allocated objects, the advantages of a [`Slab`] are lost as the memory allocated for the
/// object is freed when the object is removed from the [`Slab`]. With a pool, we can instead re-use
```

(and add a reference to the `Slab` link at the end of the doc comment)